src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -291,15 +291,54 @@ private bool StartCore(ProcessStartInfo startInfo)
                 }
             }
 
-            int childPid, stdinFd, stdoutFd, stderrFd;
+            int stdinFd = -1, stdoutFd = -1, stderrFd = -1;
             string[] envp = CreateEnvp(startInfo);
             string cwd = !string.IsNullOrWhiteSpace(startInfo.WorkingDirectory) ? startInfo.WorkingDirectory : null;
 
+            bool setCredentials = !string.IsNullOrEmpty(startInfo.UserName);
+            uint userId = 0;
+            uint groupId = 0;
+            if (setCredentials)
+            {
+                (userId, groupId) = GetUserAndGroupIds(startInfo);
+            }
+
             if (startInfo.UseShellExecute)
             {
+                if (!IsValidVerb(startInfo.Verb))
+                {
+                    throw new Win32Exception(Interop.Errors.ERROR_NO_ASSOCIATION);
+                }
+
```

Comment:
Remove
https://github.com/dotnet/corefx/blob/a11365cdab5108cc2d435322db0595e04145b4cd/src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs#L308-L311
Add
```suggestion
                string Verb = startInfo.Verb;
                if (null==Verb || ""==Verb || Verb.ToLower()=="open"){
```